test/e2e/gateway_api_test.go

```diff
@@ -423,6 +428,284 @@ func testGatewayAPIRBAC(t *testing.T) {
 	}
 }
 
+func testGatewayAPIDNS(t *testing.T) {
+	domain := "gws." + dnsConfig.Spec.BaseDomain
+
+	gatewayClass, err := createGatewayClass(operatorcontroller.OpenShiftDefaultGatewayClassName, operatorcontroller.OpenShiftGatewayClassControllerName)
+	if err != nil {
+		t.Fatalf("failed to create gatewayclass: %v", err)
+	}
+
+	testCases := []struct {
+		name           string
+		createGateways []struct {
+			gatewayName string
+			namespace   string
+			hostname    *string
+		}
+		expectedListenerConditions []metav1.Condition
+		expectedDNSRecords         map[expectedDnsRecord]bool
+	}{
+		// TODO: In this case Gateway Listeners should be reported as conflicted. To be fixed in the future release.
+		{
+			name: "multipleGatewaysSameListenerHostname",
+			createGateways: []struct {
+				gatewayName string
+				namespace   string
+				hostname    *string
+			}{
+				{gatewayName: "gw1", namespace: operatorcontroller.DefaultOperandNamespace, hostname: ptr.To("abc." + domain)},
+				{gatewayName: "gw2", namespace: operatorcontroller.DefaultOperandNamespace, hostname: ptr.To("abc." + domain)},
+			},
+			expectedListenerConditions: []metav1.Condition{
+				{Type: "Accepted", Status: metav1.ConditionTrue},
+				{Type: "Conflicted", Status: metav1.ConditionFalse},
+				{Type: "Programmed", Status: metav1.ConditionTrue},
+				{Type: "ResolvedRefs", Status: metav1.ConditionTrue},
+			},
+			expectedDNSRecords: map[expectedDnsRecord]bool{
+				{dnsName: "abc." + domain + ".", gatewayName: "gw1"}: true,
+				{dnsName: "abc." + domain + ".", gatewayName: "gw2"}: true,
```

Comment:
So, it will be the same DNS record in the same zone. Which gateway's target will win then (on AWS, on GCP, on Azure)? I saw [this thread](https://redhat-internal.slack.com/archives/CBWMXQJKD/p1743690816760349). What is the conclusion? Do we accept this a valid GatewayAPI (`Conflicted=False`) and buggy DNS management in CIO? Or we consider the latter as valid too (I would have a question about it then)?

Reply:
See the comment: 
`// TODO: In this case Gateway Listeners should be reported as conflicted. To be fixed in the future release.`
This is due to Istio, and it will need to be documented as an Istio-ism.

Reply:
I think that there are 2 sides to this thing: Istio (`Conflicted` listener status) and cluster ingress operator (`DNSRecord`). Do we consider the `DNSRecord` logic valid? On AWS, for instance, two or more gateways will be overriding CNAME record. Is this the behavior we expect?

Reply:
Yes, we should open a bug because the dnsRecord shouldn't be created for a Listener that is Conflicted=True.